internal/attestation/untyped.go

```diff
@@ -0,0 +1,161 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package attestation
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/sigstore/cosign/v2/pkg/oci"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/enterprise-contract/ec-cli/internal/signature"
+	"github.com/enterprise-contract/ec-cli/internal/utils"
+)
+
+// Some important keys we're hoping to find in the opaque data
+const (
+	predicateTypeKey = "predicateType"
+	predicateKey     = "predicate"
+)
+
+// Similar function to in_toto.ProvenanceStatementSLSA02 but we'll use
+// an opaque untyped map
+type UntypedAttestation map[string]any
```

Comment:
Could  this be an array in some cases?

Reply:
It's a good question. I think probably no, but I'm not certain.